microsoft/testsuites/core/dns.py

```diff
@@ -40,6 +41,28 @@ def verify_dns_name_resolution(self, node: Node) -> None:
     )
     def verify_dns_name_resolution_after_upgrade(self, node: Node) -> None:
         self._check_dns_name_resolution(node)
+        self._upgrade_system(node)
+        self._check_dns_name_resolution(node)
+        node.reboot()
+        self._check_dns_name_resolution(node)
+
+    @retry(tries=10, delay=0.5)
+    def _check_dns_name_resolution(self, node: Node) -> None:
+        ping = node.tools[Ping]
+        ping.ping(target="bing.com")
+
+    def _upgrade_system(self, node: Node) -> None:
+        if not isinstance(node.os, Posix):
+            raise UnsupportedDistroException(node.os)
+
+        try:
+            node.os.update_packages("")
+        except ReleaseEndOfLifeException as identifier:
+            # For the EOL release, collect the except information and skip upgrading
+            # system.
+            node.log.debug(f"{identifier}")
```

Comment:
Use PassedException here, because it's partial passed, not fully. It's like smoke test, when it cannot be rebooted.

Reply:
Done

Reply:
node.log.debug(identifier) is enough

Reply:
Done